test/cql-pytest/test_tablets.py

```diff
@@ -201,3 +201,34 @@ def test_tablets_are_dropped_when_dropping_index(cql, test_keyspace, drop_index)
         except:
             pass
         raise e
+
+# We want to ensure that we can only change the RF of any DC by at most 1 at a time
+# if we use tablets. That provides us with the guarantee that the old and the new QUORUM
+# overlap by at least one node.
+def test_alter_tablet_keyspace(cql, this_dc):
+    with new_test_keyspace(cql, f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', '{this_dc}' : 1 }} "
+                                f"AND TABLETS = {{ 'enabled': true, 'initial': 128 }}") as keyspace:
+        def change_opt_rf(rf_opt, new_rf):
+            cql.execute(f"ALTER KEYSPACE {keyspace} WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', '{rf_opt}' : {new_rf} }}")
+        def change_dc_rf(new_rf):
+            change_opt_rf(this_dc, new_rf)
+        def change_default_rf(new_rf):
+            change_opt_rf("replication_factor", new_rf)
+
+        change_dc_rf(2)
+        change_dc_rf(3)
+
+        with pytest.raises(InvalidRequest):
+            change_dc_rf(5)
+        with pytest.raises(InvalidRequest):
+            change_dc_rf(1)
+        with pytest.raises(InvalidRequest):
+            change_dc_rf(10)
+
+        # Changing the default replication factor doesn't affect any of the existing DCs,
+        # so we can change it arbitrarily.
```

Comment:
but it will affect a new DC if we add nodes to it and then issue ALTER KEYSPACE with the default replication_factor option.

Reply:
A note to myself: whether to support this tag or drop it is under discussion.

Reply:
`replication_factor` that will be rejected by alter ks with tablets, and these checks have been removed altogether.